compiler/noirc_frontend/src/hir/resolution/resolver.rs

```diff
@@ -96,6 +96,9 @@ pub struct Resolver<'a> {
     /// Used to link items to their dependencies in the dependency graph
     current_item: Option<DependencyId>,
 
+    /// In-resolution names
+    resolving_names: BTreeSet<Ident>,
```

Comment:
We should use the struct id instead of the name directly. It'd be cheaper to copy/compare and wouldn't lead to errors if the user used a field of the same name but from a different module.